app/app.go

```diff
@@ -584,6 +586,27 @@ func (app *TerraApp) Name() string { return app.BaseApp.Name() }
 
 // BeginBlocker application updates every begin block
 func (app *TerraApp) BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock) abci.ResponseBeginBlock {
+
+	// Disable IBC Channel 1 to Osmosis
+	if ctx.ChainID() == core.ColumbusChainID && ctx.BlockHeight() == core.SwapDisableForkHeight {
+		channel, found := app.IBCKeeper.ChannelKeeper.GetChannel(ctx, ibctransfertypes.PortID, "channel-1")
+		if !found {
+			panic("channel-1 not found")
+		}
+		channel.State = ibcchanneltypes.CLOSED
```

Comment:
This allows the channel to be closed on Osmosis as well (via ChanCloseConfirm). This might be okay, but to recover it on Osmosis would require similar code to this. You could prevent the closure on osmosis by changing one of the other channel fields (such as version or counterparty hops). The version might make sense, you could just change it to "ics20-1-disabled", but perhaps this is not worth increasing the scope of hacked in changes

In the future, we should make available a middleware for disabling certain channels.

Reply:
Changing the channel fields would prevent timeouts from occurring (on the terra side), so I'd actually avoid that. The cleanest solution would be to use a fork of ibc-go and add the check [here](https://github.com/cosmos/ibc-go/blob/9f70a070d773f8bfdb62d4205d8878f3149f351a/modules/apps/transfer/keeper/relay.go#L62). That way you don't adjust the channel state (and cause unwanted sideaffects), but still disable sends

Reply:
Oh actually, I'd just add an [ante handler](https://github.com/cosmos/ibc-go/blob/39a6f5fdc862d83b0ad0cdf6780fd143383cc0cc/modules/core/ante/ante.go#L22) for this. If the msg is a MsgTransfer and the source port/channel match the ones above, then return an error. This should have the same desired affect?

Reply:
I think it is fine that the channel gets closed on Osmosis as well, and it requires a manual intervention on both ends to reopen.

Also if you disable sends, you will disable sending for all channels which doesn't seem to be the intent of this PR which only closes the channels to DEX chains.

This will allow relayers to still process timeouts so pending sends can be refunded to original users. This may require relayers to support the `TimeoutOnCloseMsg` but it is not time sensitive per se.

Any cross chain funds (ie. Terra on Osmosis and Osmo on Terra) will be stuck on the remote chain until/unless the channel reopens.

Reply:
@colin-axner does no channel close on Osmosis mean UST can be withdrawn back to Terra?